hooks/pre-checkout

```diff
@@ -2,7 +2,30 @@
 
 set -euo pipefail
 
+# Prints an info line to stdout.
+log_info() {
+  echo "$(date '+[%Y-%m-%d %H:%M:%S]') INFO: $*"
+}
+
+export_canva_origin_url() {
+  local origin_url
+  local s3_path
+  log_info "calling SSM param"
+  origin_url="$(aws ssm get-parameter \
+      --name "${BUILDKITE_PLUGIN_GITHUB_FETCH_SSM_PARAM}"\
+      --query "Parameter.Value"\
+      --output text\
+      --region "us-east-1")"
+  base="$(basename ${origin_url})"
+  repo_dir="${base%.*}"
+  if [[ -n "${BUILDKITE_PLUGIN_GITHUB_FETCH_S3_URL:-}" ]]; then
+    export BUILDKITE_PLUGIN_GITHUB_FETCH_S3_URL="${BUILDKITE_PLUGIN_GITHUB_FETCH_S3_URL}/${repo_dir}"
```

Comment:
changed this, since this should point to `s3://build.canva-build.com/checkout/canva` not `s3://build.canva-build.com/checkout/git_github_com_Canva_canva_git/`. This is because we already have a different checkout path for the infrastructure repo, and it's extrapolated to be `s3://build.canva-build.com/checkout/infrastructure`. No real reason to change this, and if we want the initial checkout path, we'd need to change all the infra repo pipelines to use it too

Reply:
cc @arromer if you have any concerns about this let me know, I don't think this necessitates another review though